components/external_image/external_image.tsx

```diff
@@ -1,20 +1,25 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import {getImageSrc} from 'utils/post_utils.jsx';
 
-export default class ExternalImage extends React.PureComponent {
-    static propTypes = {
-        children: PropTypes.func.isRequired,
-        enableSVGs: PropTypes.bool.isRequired,
-        hasImageProxy: PropTypes.bool.isRequired,
-        imageMetadata: PropTypes.object,
-        src: PropTypes.string.isRequired,
-    };
+interface ImageMetaData {
+    format?: string;
+    width?: number;
+    height?: number;
+}
+
+interface Props {
+    children: (src : string) => React.ReactNode;
+    enableSVGs: boolean;
+    hasImageProxy: boolean;
+    imageMetadata?: ImageMetaData,
```

Comment:
Import `PostImage` from redux and use it here.
You can create a new PR which adds those above fields to the `PostImage` object in the redux repository.

Reply:
I changed my code and pushed again. Actually, I changed types as you advised me in the last comment. I checked type-test locally and it worked okay. But the problem is I changed types in mattermost-redux which is belongs to node_modules. Hence, its changes has not been updated in this commit and type-checks errors are keep failing.

Reply:
I requested PR in redux-repository and waiting for the merge.

Reply:
The build failure is because the redux hash needs to be update with latest commit in your redux PR. 
Once your redux PR gets merged, then we can update the redux hash with the latest in master redux before merging this PR into master.